src/main/java/com/fasterxml/jackson/databind/deser/SettableAnyProperty.java

```diff
@@ -437,4 +459,49 @@ public SettableAnyProperty withValueDeserializer(JsonDeserializer<Object> deser)
             return this;
         }
     }
+
+    /**
+     * @since 2.18
+     */
+    protected static class MapParameterAnyProperty extends SettableAnyProperty
+        implements java.io.Serializable
+    {
+        private static final long serialVersionUID = 1L;
+
+        protected final ValueInstantiator _valueInstantiator;
+
+        public MapParameterAnyProperty(BeanProperty property,
+                                       AnnotatedMember field, JavaType valueType,
+                                       KeyDeserializer keyDeser,
+                                       JsonDeserializer<Object> valueDeser, TypeDeserializer typeDeser,
+                                       ValueInstantiator inst) {
+            super(property, field, valueType,
+                keyDeser, valueDeser, typeDeser);
+            _valueInstantiator = inst;
+        }
+
+        @Override
+        public SettableAnyProperty withValueDeserializer(JsonDeserializer<Object> deser) {
+            return new MapParameterAnyProperty(_property, _setter, _type,
+                _keyDeserializer, deser, _valueTypeDeserializer,
+                _valueInstantiator);
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        protected void _set(Object instance, Object propName, Object value) throws Exception {
+            throw new UnsupportedOperationException("Cannot set any properties for constructor parameter of type `Map`");
+        }
+
+        protected Map<Object, Object> initMap(DeserializationContext ctxt)
+            throws IOException
+        {
+            if (_valueInstantiator == null) {
```

Comment:
It seems to me that this really should be caught during construction and never get this far?

Reply:
Moved the check accordingly.